src/main/java/org/jsmart/zerocode/core/httpclient/ssl/CorporateProxyNoSslContextHttpClient.java

```diff
@@ -0,0 +1,203 @@
+package org.jsmart.zerocode.core.httpclient.ssl;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CookieStore;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.entity.EntityBuilder;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.jsmart.zerocode.core.httpclient.BasicHttpClient;
+import org.jsmart.zerocode.core.utils.HelperJsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.core.Response;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+import static org.jsmart.zerocode.core.utils.HelperJsonUtils.getContentAsItIsJson;
+
+public class CorporateProxyNoSslContextHttpClient implements BasicHttpClient {
```

Comment:
cover with test and create a sample please!

Reply:
you need a actual proxy or a mock proxy to test this. Will do this later. Issue #63 has been raised to address this.